include/beast/http/impl/message_v1.ipp

```diff
@@ -139,9 +140,35 @@ prepare(message_v1<isRequest, Body, Headers>& msg,
     {
         if(pi.content_length)
         {
-            // VFALCO TODO Use a static string here
-            msg.headers.insert("Content-Length",
-                std::to_string(*pi.content_length));
+            struct set_field
+            {
+                void
+                operator()(message_v1<true, Body, Headers>& msg,
+                    detail::prepare_info const& pi) const
+                {
+                    using beast::detail::ci_equal;
+                    if(*pi.content_length > 0 ||
+                        ci_equal(msg.method, "POST"))
+                    {
+                        msg.headers.insert("Content-Length",
+                            std::to_string(*pi.content_length));
+                    }
+                }
+
+                void
+                operator()(message_v1<false, Body, Headers>& msg,
+                    detail::prepare_info const& pi) const
+                {
+                    if((msg.status / 100 ) != 1 &&
+                        msg.status != 204 &&
+                        msg.status != 304)
+                    {
+                        msg.headers.insert("Content-Length",
+                            std::to_string(*pi.content_length));
+                    }
+                }
+            };
+            set_field{}(msg, pi);
```

Comment:
I like this idiom. It's an interesting way to give us overloaded local functions when we don't have `if constexpr` or overloaded lambdas. What do you think about renaming the struct to `overloads` and having static overloads called `set_field` to make it clear what the idiom is doing? Too verbose?
```            
            struct overloads
            {
                static
                void
                set_field(message_v1<true, Body, Headers>& msg,
                    detail::prepare_info const& pi)
                {
                    using beast::detail::ci_equal;
                    if(*pi.content_length > 0 ||
                        ci_equal(msg.method, "POST"))
                    {
                        msg.headers.insert("Content-Length",
                            std::to_string(*pi.content_length));
                    }
                }

                static
                void
                set_field(message_v1<false, Body, Headers>& msg,
                    detail::prepare_info const& pi)
                {
                    if((msg.status / 100 ) != 1 &&
                        msg.status != 204 &&
                        msg.status != 304)
                    {
                        msg.headers.insert("Content-Length",
                            std::to_string(*pi.content_length));
                    }
                }
            };
            overloads::set_field(msg, pi);
```